bin/bench-tx/src/main.rs

```diff
@@ -110,8 +110,13 @@ pub fn benchmark_p2id() -> Result<TransactionMeasurements, String> {
     )
     .unwrap();
 
+    let tx_script_target =
+        TransactionScript::compile(DEFAULT_AUTH_SCRIPT, [], TransactionKernel::assembler())
+            .unwrap();
+
     let tx_context = TransactionContextBuilder::new(target_account.clone())
         .input_notes(vec![note.clone()])
+        .tx_script(tx_script_target.clone())
         .build();
```

Comment:
@igamigo @PhilippGackstatter - the fact that we need to use transaction script twice (once for `tx_context` and again for `tx_args` online 126 below) feels a bit awkward to me. I'm guessing this is because with the current structure all programs must be pre-loaded into a MAST forest store before we can execute a transaction.

If so, I wonder if we should introduce some kind of an additional layer in the `TransactionExecutor` so that it could have a local MAST forest store but also request additional data from the data store if needed.

Reply:
It seems to me the problem here is more that we're passing everything required into the transaction context builder, and then instead of executing it, we get out all the parts required to pass to the transaction executor. This is essentially what `TransactionContext::execute` does, so I've refactored it in #1354.